### PR TITLE
e2e_network: stop using deprecated framework.ExpectEqual

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

e2e_network: stop using deprecated framework.ExpectEqual

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/pull/115961

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by danwinship** [comment]
/approve
I'll leave it to someone more sig-testing-y to confirm that this implements the sig-testing plan properly and lgtm it.

**Comment by aojea** [comment]
Patrick is leading this effort, he is the best one to judge

/assign @pohly

File: test/e2e/network/endpointslice.go [hunk 1]
```diff
@@ -462,7 +463,7 @@ var _ = common.SIGDescribe("EndpointSlice", func() {
 		ginkgo.By("patching")
 		patchedEPS, err := epsClient.Patch(ctx, createdEPS.Name, types.MergePatchType, []byte(`{"metadata":{"annotations":{"patched":"true"}}}`), metav1.PatchOptions{})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(patchedEPS.Annotations["patched"], "true", "patched object should have the applied annotation")
+		gomega.Expect(patchedEPS.Annotations["patched"]).To(gomega.Equal("true"), "patched object should have the applied annotation")
```
Comment (pohly) [inline test/e2e/network/endpointslice.go:466]: Better use `HaveKeyWithValue` here.

File: test/e2e/network/endpointslice.go [hunk 2]
```diff
@@ -476,7 +477,7 @@ var _ = common.SIGDescribe("EndpointSlice", func() {
 			return err
 		})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(updatedEPS.Annotations["updated"], "true", "updated object should have the applied annotation")
+		gomega.Expect(updatedEPS.Annotations["updated"]).To(gomega.Equal("true"), "updated object should have the applied annotation")
```
Comment (pohly) [inline test/e2e/network/endpointslice.go:480]: And here.

File: test/e2e/network/ingress.go [hunk 3]
```diff
@@ -703,7 +704,7 @@ var _ = common.SIGDescribe("Ingress API", func() {
 		ginkgo.By("patching")
 		patchedIngress, err := ingClient.Patch(ctx, createdIngress.Name, types.MergePatchType, []byte(`{"metadata":{"annotations":{"patched":"true"}}}`), metav1.PatchOptions{})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(patchedIngress.Annotations["patched"], "true", "patched object should have the applied annotation")
+		gomega.Expect(patchedIngress.Annotations["patched"]).To(gomega.Equal("true"), "patched object should have the applied annotation")
```
Comment (pohly) [inline test/e2e/network/ingress.go:707]: And here.

File: test/e2e/network/ingress.go [hunk 4]
```diff
@@ -717,7 +718,7 @@ var _ = common.SIGDescribe("Ingress API", func() {
 			return err
 		})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(updatedIngress.Annotations["updated"], "true", "updated object should have the applied annotation")
+		gomega.Expect(updatedIngress.Annotations["updated"]).To(gomega.Equal("true"), "updated object should have the applied annotation")
```
Comment (pohly) [inline test/e2e/network/ingress.go:721]: And here.

File: test/e2e/network/ingress.go [hunk 5]
```diff
@@ -754,8 +755,8 @@ var _ = common.SIGDescribe("Ingress API", func() {
 			[]byte(`{"metadata":{"annotations":{"patchedstatus":"true"}},"status":{"loadBalancer":`+string(lbStatusJSON)+`}}`),
 			metav1.PatchOptions{}, "status")
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(patchedStatus.Status.LoadBalancer, lbStatus, "patched object should have the applied loadBalancer status")
-		framework.ExpectEqual(patchedStatus.Annotations["patchedstatus"], "true", "patched object should have the applied annotation")
+		gomega.Expect(patchedStatus.Status.LoadBalancer).To(gomega.Equal(lbStatus), "patched object should have the applied loadBalancer status")
+		gomega.Expect(patchedStatus.Annotations["patchedstatus"]).To(gomega.Equal("true"), "patched object should have the applied annotation")
```
Comment (pohly) [inline test/e2e/network/ingress.go:759]: And here.

File: test/e2e/network/ingressclass.go [hunk 6]
```diff
@@ -350,14 +351,14 @@ var _ = common.SIGDescribe("IngressClass API", func() {
 		ginkgo.By("patching")
 		patchedIC, err := icClient.Patch(ctx, ingressClass1.Name, types.MergePatchType, []byte(`{"metadata":{"annotations":{"patched":"true"}}}`), metav1.PatchOptions{})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(patchedIC.Annotations["patched"], "true", "patched object should have the applied annotation")
+		gomega.Expect(patchedIC.Annotations["patched"]).To(gomega.Equal("true"), "patched object should have the applied annotation")
```
Comment (pohly) [inline test/e2e/network/ingressclass.go:354]: And here.

File: test/e2e/network/ingressclass.go [hunk 7]
```diff
@@ -350,14 +351,14 @@ var _ = common.SIGDescribe("IngressClass API", func() {
...
 
 		ginkgo.By("updating")
 		icToUpdate := patchedIC.DeepCopy()
 		icToUpdate.Annotations["updated"] = "true"
 		updatedIC, err := icClient.Update(ctx, icToUpdate, metav1.UpdateOptions{})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(updatedIC.Annotations["updated"], "true", "updated object should have the applied annotation")
+		gomega.Expect(updatedIC.Annotations["updated"]).To(gomega.Equal("true"), "updated object should have the applied annotation")
```
Comment (pohly) [inline test/e2e/network/ingressclass.go:361]: And here.

File: test/e2e/network/netpol/network_policy_api.go [hunk 8]
```diff
@@ -159,14 +160,14 @@ var _ = common.SIGDescribe("Netpol API", func() {
 		ginkgo.By("patching")
 		patchedNetPols, err := npClient.Patch(ctx, createdNetPol.Name, types.MergePatchType, []byte(`{"metadata":{"annotations":{"patched":"true"}}}`), metav1.PatchOptions{})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(patchedNetPols.Annotations["patched"], "true", "patched object should have the applied annotation")
+		gomega.Expect(patchedNetPols.Annotations["patched"]).To(gomega.Equal("true"), "patched object should have the applied annotation")
```
Comment (pohly) [inline test/e2e/network/netpol/network_policy_api.go:163]: And here.

File: test/e2e/network/netpol/network_policy_api.go [hunk 9]
```diff
@@ -159,14 +160,14 @@ var _ = common.SIGDescribe("Netpol API", func() {
...
 
 		ginkgo.By("updating")
 		npToUpdate := patchedNetPols.DeepCopy()
 		npToUpdate.Annotations["updated"] = "true"
 		updatedNetPols, err := npClient.Update(ctx, npToUpdate, metav1.UpdateOptions{})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(updatedNetPols.Annotations["updated"], "true", "updated object should have the applied annotation")
+		gomega.Expect(updatedNetPols.Annotations["updated"]).To(gomega.Equal("true"), "updated object should have the applied annotation")
```
Comment (pohly) [inline test/e2e/network/netpol/network_policy_api.go:170]: And here.

File: test/e2e/network/service.go [hunk 10]
```diff
@@ -3231,7 +3231,7 @@ var _ = common.SIGDescribe("Services", func() {
 		ginkgo.By("fetching the Endpoint")
 		endpoints, err := f.ClientSet.CoreV1().Endpoints(testNamespaceName).Get(ctx, testEndpointName, metav1.GetOptions{})
 		framework.ExpectNoError(err, "failed to fetch Endpoint")
-		framework.ExpectEqual(foundEndpoint.ObjectMeta.Labels["test-service"], "updated", "failed to update Endpoint %v in namespace %v label not updated", testEndpointName, testNamespaceName)
+		gomega.Expect(foundEndpoint.ObjectMeta.Labels["test-service"]).To(gomega.Equal("updated"), "failed to update Endpoint %v in namespace %v label not updated", testEndpointName, testNamespaceName)
```
Comment (pohly) [inline test/e2e/network/service.go:3234]: And here.

File: test/e2e/network/service.go [hunk 11]
```diff
@@ -3279,13 +3279,13 @@ var _ = common.SIGDescribe("Services", func() {
 		ginkgo.By("fetching the Endpoint")
 		endpoints, err = f.ClientSet.CoreV1().Endpoints(testNamespaceName).Get(ctx, testEndpointName, metav1.GetOptions{})
 		framework.ExpectNoError(err, "failed to fetch Endpoint")
-		framework.ExpectEqual(endpoints.ObjectMeta.Labels["test-service"], "patched", "failed to patch Endpoint with Label")
+		gomega.Expect(endpoints.ObjectMeta.Labels["test-service"]).To(gomega.Equal("patched"), "failed to patch Endpoint with Label")
```
Comment (pohly) [inline test/e2e/network/service.go:3282]: And here.

File: test/e2e/network/service.go [hunk 12]
```diff
@@ -3918,7 +3918,7 @@ func execAffinityTestForSessionAffinityTimeout(ctx context.Context, f *framework
 	framework.ExpectNoError(err)
 
 	// the service should be sticky until the timeout expires
-	framework.ExpectEqual(checkAffinity(ctx, cs, execPod, svcIP, servicePort, true), true)
+	gomega.Expect(checkAffinity(ctx, cs, execPod, svcIP, servicePort, true)).To(gomega.BeTrue())
```
Comment (pohly) [inline test/e2e/network/service.go:3921]: Can you also change this to not use `BeTrue`? See https://github.com/kubernetes/kubernetes/issues/105678.

File: test/e2e/network/service.go [hunk 13]
```diff
@@ -3999,19 +3999,19 @@ func execAffinityTestForNonLBServiceWithOptionalTransition(ctx context.Context,
 	framework.ExpectNoError(err)
 
 	if !isTransitionTest {
-		framework.ExpectEqual(checkAffinity(ctx, cs, execPod, svcIP, servicePort, true), true)
+		gomega.Expect(checkAffinity(ctx, cs, execPod, svcIP, servicePort, true)).To(gomega.BeTrue())
```
Comment (pohly) [inline test/e2e/network/service.go:4002]: And here.

File: test/e2e/network/service.go [hunk 14]
```diff
@@ -3999,19 +3999,19 @@ func execAffinityTestForNonLBServiceWithOptionalTransition(ctx context.Context,
...
 	}
 	if isTransitionTest {
 		_, err = jig.UpdateService(ctx, func(svc *v1.Service) {
 			svc.Spec.SessionAffinity = v1.ServiceAffinityNone
 		})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(checkAffinity(ctx, cs, execPod, svcIP, servicePort, false), true)
+		gomega.Expect(checkAffinity(ctx, cs, execPod, svcIP, servicePort, false)).To(gomega.BeTrue())
```
Comment (pohly) [inline test/e2e/network/service.go:4009]: And here.

File: test/e2e/network/service.go [hunk 15]
```diff
@@ -4049,19 +4049,19 @@ func execAffinityTestForLBServiceWithOptionalTransition(ctx context.Context, f *
 	port := int(svc.Spec.Ports[0].Port)
 
 	if !isTransitionTest {
-		framework.ExpectEqual(checkAffinity(ctx, cs, nil, ingressIP, port, true), true)
+		gomega.Expect(checkAffinity(ctx, cs, nil, ingressIP, port, true)).To(gomega.BeTrue())
```
Comment (pohly) [inline test/e2e/network/service.go:4052]: And here.

File: test/e2e/network/service.go [hunk 16]
```diff
@@ -4049,19 +4049,19 @@ func execAffinityTestForLBServiceWithOptionalTransition(ctx context.Context, f *
...
 	}
 	if isTransitionTest {
 		svc, err = jig.UpdateService(ctx, func(svc *v1.Service) {
 			svc.Spec.SessionAffinity = v1.ServiceAffinityNone
 		})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(checkAffinity(ctx, cs, nil, ingressIP, port, false), true)
+		gomega.Expect(checkAffinity(ctx, cs, nil, ingressIP, port, false)).To(gomega.BeTrue())
```
Comment (pohly) [inline test/e2e/network/service.go:4059]: And here.

File: test/e2e/network/service.go [hunk 17]
```diff
@@ -4049,19 +4049,19 @@ func execAffinityTestForLBServiceWithOptionalTransition(ctx context.Context, f *
...
 		svc, err = jig.UpdateService(ctx, func(svc *v1.Service) {
 			svc.Spec.SessionAffinity = v1.ServiceAffinityClientIP
 		})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(checkAffinity(ctx, cs, nil, ingressIP, port, true), true)
+		gomega.Expect(checkAffinity(ctx, cs, nil, ingressIP, port, true)).To(gomega.BeTrue())
```
Comment (pohly) [inline test/e2e/network/service.go:4064]: And here.

**Review by pohly — COMMENTED** [review]
Looks good, some suggestions.

**Comment by carlory** [comment]
@pohly please review again.

File: test/e2e/network/service.go (hunk 12 above)
Comment (pohly) [reply test/e2e/network/service.go:3921]: This isn't resolved?

Comment (carlory) [reply test/e2e/network/service.go]: @pohly Sorry, it was resolved but I forgot remove this line.

**Review by pohly — APPROVED** [review]
/lgtm
/approve

IMHO this PR confirms that deprecating framework.ExpectEqual was the right choice: several of the revised checks are now more expressive, with better failure messages if things go wrong. I also find the checks more readable.

Thanks @carlory !

**Pull request merged**